### DISCUSSION TITLE
How to access built-in directives?

### DISCUSSION THREAD

**Discussion by spasarok** [graph-gophers/graphql-go#652] [Q&A] [answered]
Hello! How do I access query values for built-in directives like `@skip` and `@include`? I cannot find a way to access these values in order to apply them to the server-side query resolution.

**Comment by pavelnikolov** [comment]
Hi @spasarok,

These built-in directives are handled by the lib automatically. You do not have to do anything on the server-side to apply them. See unit tests, for example, [TestSkipDirective](https://github.com/graph-gophers/graphql-go/blob/master/graphql_test.go#L1958C6-L1958C23) and [TestIncludeDirective](https://github.com/graph-gophers/graphql-go/blob/master/graphql_test.go#L2023C6-L2023C26).

**Reply by spasarok** [reply]
@pavelnikolov Thanks for the response. I see that when `@skip` is applied to a field, the field is omitted from the response data returned to the user. However, what I don't understand is how to actually prevent the server-side logic from hitting the database if a field is skipped. From what I see, I need access to the boolean value of `@skip` if I want to prevent my server-side logic from hitting the database when a field is skipped (and this is what I can't figure out how to do). Can you clarify?

**Reply by pavelnikolov** [reply]
To be honest, this logic predates me in this project and has been like that ever since. I am not really sure about the details and cannot check right now but you can easily experiment with that - just create a simple resolver with which resolves a filed, then make sure that in the corresponding resolver you log a message every time it is selected. This will prove if your theory is right or wrong.

**Reply by spasarok** [reply]
@pavelnikolov Thanks for the response. I've already tested this and reviewed the source code, and it looks like the library doesn't support this. My discussion question was mostly to confirm that I'm not missing something. Are you open to PRs?

**Reply by pavelnikolov** [reply]
PRs are welcome.

**Reply by spasarok** [reply]
Awesome. I don't have the bandwidth to implement this right now, but I'll look into this if I find the time. It would be a useful performance gain for my project. Thank you!

**Reply by spasarok** [reply, marked as answer]
~In conclusion, the module does handle omitting fields marked with `@skip` from the response structure returned to the user. However, I cannot currently find a way to prevent the server from still processing fields marked with `@skip`. If I do find a way to achieve this, I will update here.~

Update: `@skip` appears to work as expected when using method resolvers. I will switch to method resolvers for fields I want to enable skipping. Thanks!
